src/Runner.Worker/Container/DockerUtil.cs

```diff
@@ -61,5 +62,11 @@ public static string ParseRegistryHostnameFromImageName(string name)
             }
             return "";
         }
+
+        public static bool IsDockerfile(string image)
+        {
+            var imageWithoutPath = image.Split('/').Last();
```

Comment:
Are we overthinking this? If we swap the order on the if statement, should we just check if the path contains "dockerfile"?

Reply:
My view is that we should not change the order of the if statement. Also, if the image is named `nikola-jokic/dockerfile:latest`, and if we check only for existence of the dockerfile, we would treat this as a file, but it is not.
We are running those dockerfiles with `docker run -f [path to file]...`. This means that in theory, we could name the file whatever we want, let's say `ContainerDeffinition` and put it inside `/dockerfile/ContainerDeffinition`, and it would work (because of the `-f` flag). But I am worried if the image name contains dockerfile, we would treat it as a file to build an image from, and we would not try to pull it from the docker hub registry.
This way, we clearly allow prefix `dockerfile.` or endswith `dockerfile`.
But if you feel like we could allow anything if the path contains dockerfile, should we check if the last part is a tag (For example `:latest` or `:20.04`)? If so, that is **most likely** not a dockerfile

Reply:
So, the changes as they are currently would skip actual local dockerfiles with the following names:

```
/foo/bar/nikola-jokic/dockerfile:latest
./dockerfile:latest
```

I don't think using contains is the way to go, testing contains(dockerfile) for the element after the last `/`, we would treat URLs like these as file paths:

```
nikola-jokic/dockerfile:latest                            # dockerhub URL
nikola-jokic/app-dockerfile-test:latest
```

I don't think we can have a perfect solution here. Probably the most user friendly way would be to try and treat `image` as a path, if it exists, build it, if not, fallback and try treating it as a URL. We have already discussed why we don't want to do that.